### PR TITLE
Added Session03 exercises

### PR DESCRIPTION
**Pull request by dtingg**
I added slicing_lab, list_lab, strformat_lab, and mailroom to the Session03 folder.

### CODE REVIEW

File: students/DiannaTingg/Session03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+for i in range(len(reverse_fruits)):
+    reverse_fruits[i] = reverse_fruits[i][::-1]
+
+# Delete the last item of the original list
+my_fruits.pop()
+
+# Print the original list and the copy.
+print("The original list is: {}".format(my_fruits))
+print("The reversed list is: {}".format(reverse_fruits))
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/list_lab.py]: Excellent!

File: students/DiannaTingg/Session03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    # Until the user selects Q for quit
+    while answer != "Q":
+        print("\nMAIN MENU")
+        print("Type T to Send a Thank You")
+        print("Type R to Create a Report")
+        print("Type Q to Quit")
+
+        answer = input("\nPlease enter a command: ")
+        answer = answer[0].capitalize()
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/mailroom.py]: Nice use of capitalize!

File: students/DiannaTingg/Session03/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+# Lesson 03 Assignment: Mailroom, Part 1
+
+# Make a list that holds donor names and a history of the amounts they donated
+donors = [("Jimmy Fallon", [500.00, 700.00]), ("Taylor Swift", [1000.00]), ("Dan White", [200.00, 300.00, 400.00]),
+          ("Trevor Noah", [750.00, 850.00]), ("Elon Musk", [3000.00]), ("Selena Gomez", [100.00, 150.00, 200.00])]
+
+
+# Display menu options
+def display_menu():
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/mailroom.py]: You can remove most or all of the blank rows to make your functions more compact.

File: students/DiannaTingg/Session03/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+
+# Send a Thank You
+def send_thank_you():
+    name = ""
+
+    while name == "" or name == "E" or name == "L":
+        name = input("Please enter the donor's full name, "
+                     "type L to see a list of previous donors, "
+                     "or type E to exit to main menu: ").title()
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/mailroom.py]: Good use of Title here

File: students/DiannaTingg/Session03/mailroom.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+def main():
+    print("Welcome to Dianna's Mailroom Program!")
+
+    display_menu()
+
+    print("Goodbye!")
+
+
+if __name__ == "__main__":
+    main()
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/mailroom.py]: Program works great - nice work!

File: students/DiannaTingg/Session03/slicing_lab.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    else:
+        return new_seq
+
+
+def exchange_first_last(seq):
+    """
+    Return sequence with first and last items exchanged.
+    """
+    # Create new list and set it to the last element of the original sequence
+    new_seq = [seq[-1]]
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/slicing_lab.py]: You'll want to use slicing for all of these functions in this exercise.

File: students/DiannaTingg/Session03/slicing_lab.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+    # Add the middle elements from the original sequence
+    new_seq.extend(seq[1:-1])
+
+    # Add the first element from the original sequence
+    new_seq.append(seq[0])
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/slicing_lab.py]: Use slicing here too.

File: students/DiannaTingg/Session03/slicing_lab.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+    test_string = "ABCDEFGHIJKL"
+    test_tuple = (1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12)
+    test_array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+
+    assert last_first_middle_third(test_string) == "IJKLABCDEFGH"
+    assert last_first_middle_third(test_tuple) == (9, 10, 11, 12, 1, 2, 3, 4, 5, 6, 7, 8)
+    assert last_first_middle_third(test_array) == [9, 10, 11, 12, 1, 2, 3, 4, 5, 6, 7, 8]
+
+    print("Tests passed!")
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/slicing_lab.py]: Nice assert statements.

File: students/DiannaTingg/Session03/strformat_lab.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+print(format_file_2(tuple_4))
+
+# Task Three
+print("\nTask Three")
+
+# Rewrite "the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3) to take an arbitrary number of values
+
+
+def formatter(in_tuple):
+    form_string = "the {} numbers are: "
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/strformat_lab.py]: Remove the extra blank rows here - not needed.

File: students/DiannaTingg/Session03/strformat_lab.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+
+def print_table_2(t):
+    """
+    Prints the tuple in columns that are 5 characters wide
+    """
+    for x in t: print(str(x).ljust(5, " "), end="")
+
+
+print_table_2(tuple_10)
```
Comment (charlesrobison) [inline students/DiannaTingg/Session03/strformat_lab.py]: Looks good!

**Comment by dtingg** [comment]
Thanks Charles!  I fixed my slicing_lab file and resubmitted it.

Regarding the extra blank lines, PyCharm is telling me that PEP 8 requires two blank lines between functions that are not within a class.

**Pull request merged**